tests/pyfunc/docker/test_docker_flavors.py

```diff
@@ -81,6 +83,48 @@ def model_path(tmp_path):
         shutil.rmtree(model_path, ignore_errors=True)
 
 
+@contextlib.contextmanager
+def start_container(port: int):
+    container = docker_client.containers.run(
+        image=TEST_IMAGE_NAME,
+        ports={8080: port},
+        detach=True,
+    )
+
+    def stream_logs():
+        for line in container.logs(stream=True):
+            sys.stdout.write(line.decode("utf-8"))
+
+    # Start a thread to stream logs from the container
+    t = threading.Thread(target=stream_logs, daemon=True)
+    t.start()
+
+    try:
+        # Wait for the server to start
+        for _ in range(30):
+            try:
+                response = requests.get(url=f"http://localhost:{port}/ping")
+                if response.ok:
+                    break
+            except requests.exceptions.ConnectionError:
+                time.sleep(5)
+
+            container.reload()
```

Comment:
what does `container.reload()` do?

Reply:
It updates the container status. If we don't call it, the status remains `"created"`.